work_queue/src/work_queue.c

```diff
@@ -271,6 +273,11 @@ struct work_queue_worker {
 										// If -1, means the worker has not reported in. If 0, means no limit.
 };
 
+struct work_queue_factory {
```

Comment:
work_queue_factory_info  so we do not confuse it with the work_queue_factory program.